packages/vm/src/testing/instance.rs

```diff
@@ -17,7 +17,8 @@ use super::storage::MockStorage;
 /// This gas limit is used in integration tests and should be high enough to allow a reasonable
 /// number of contract executions and queries on one instance. For this reason it is significatly
 /// higher than the limit for a single execution that we have in the production setup.
-const DEFAULT_GAS_LIMIT: u64 = 500_000_000; // ~0.5ms
+//const DEFAULT_GAS_LIMIT: u64 = 500_000_000; // ~0.5ms
+const DEFAULT_GAS_LIMIT: u64 = 2_150_000_000; // ~2.15ms
```

Comment:
I'm not happy about this. This is because the pairing equality itself took ~1.7ms in execution time on my machine, and then hashing to G2 takes another 392µs, so getting into an execution-time of under 2ms is very unlikely (well, at least with the current BLS12-381 library we chose)

Reply:
I don't think there is anything wrong with 2ms. However, two thoughts:
- I'd like to change the BLS implementation from zk_crypto to arkworks as pairing form zk_crypto seems to be unmaintained. This is a pure-Rust implementation. See https://github.com/noislabs/drand-verify/pull/26/files for a diff on how to use this one.
- A lot of folks recommend [blst](https://github.com/supranational/blst) as being the fastest implementation. We can use it in the VM but it is implemented in C. Not sure if the performance gain is worth the extra build system complexity. I never tried that one.

Reply:
For now I'll just port it to an arkworks backend. Maybe we can try blst later on.